python-checks/src/main/java/org/sonar/python/checks/UnusedFunctionParameterCheck.java

```diff
@@ -102,7 +104,18 @@ private static boolean isException(SubscriptionContext ctx, FunctionDef function
       !functionDef.decorators().isEmpty() ||
       isSpecialMethod(functionDef) ||
       hasNonCallUsages(functionSymbol) ||
-      isTestFunction(ctx, functionDef);
+      isTestFunction(ctx, functionDef) ||
+      isAbstractClass(functionDef);
+  }
+
+  private static boolean isAbstractClass(FunctionDef functionDef) {
+    FunctionSymbol functionSymbol = ((FunctionDefImpl) functionDef).functionSymbol();
+    if (functionSymbol == null) {
```

Comment:
To be noted that you have to do this defensive check because `FunctionDefImpl::functionSymbol` can, in theory, be `null`, but I don't think it can actually happen in practice. 
This is why you won't be able to have 100% code coverage here. One trick would be to wrap this into an `Optional` similarly to what `hasNonCallUsages` is doing, but the implications of that have to be understood and shouldn't be abused (e.g, the `null` case still won't be tested despite 100% coverage).

(No action required, just an FYI)